tests/integration/settings/user-settings/change-name-settings.test.ts

```diff
@@ -67,6 +68,22 @@ describe("ChangeNameSettings", () => {
         expect(mockCall).toHaveBeenCalled();
     });
 
+    test("When the button is clicked with the same name, then the api is not called", async () => {
+        const mockCall = mockApiCall("updateUser", Users.johnDoe);
+
+        render(ChangeNameSettings, { context: mockUserContext });
+
+        const renameButton = screen.getByRole("button", { name: "Rename" });
+
+        await userEvent.click(renameButton);
+        expect(mockCall).not.toHaveBeenCalled();
+        expect(
+            screen.getByText(
+                "To successfully change your name, you must provide a new one that is different from your current one.",
+            ),
+        ).toBeInTheDocument();
```

Comment:
```suggestion
        expect(
            screen.getByRole("heading", {
                name: "To successfully change your name, you must provide a new one that is different from your current one.",
            }),
        ).toBeInTheDocument();
```

Why not like this? The test below this one is testing for the error alert in this way.

Reply:
I changed it to use the alert role for both tests.